lib/test_queue/runner/rspec3.rb

```diff
@@ -10,23 +10,42 @@ def example_groups
       @world.ordered_example_groups
     end
 
-    def run_specs(iterator)
-      @configuration.reporter.report(@world.ordered_example_groups.count) do |reporter|
-        begin
-          hook_context = SuiteHookContext.new
-          @configuration.hooks.run(:before, :suite, hook_context)
+    # RSpec 3.2+
+    if ::RSpec::Core::Configuration.method_defined?(:with_suite_hooks)
+      def run_specs(iterator)
```

Comment:
I would prefer not to have two separate implementions of `run_specs` in this file.

Maybe we can monkey-patch a backwards compatible `with_suite_hooks` into the older version of rspec?

``` ruby
unless ::RSpec::Core::Configuration.method_defined?(:with_suite_hooks)
  class RSpec::Core::Configuration
    def with_suite_hooks
      hook_context = SuiteHookContext.new
      @configuration.hooks.run(:before, :suite, hook_context)
      yield
    ensure
      @configuration.hooks.run(:after, :suite, hook_context)
    end
  end
end
```